src/util/integer_cln_imp.h

```diff
@@ -74,8 +74,8 @@ class Integer
   Integer(unsigned long int z) : d_value(z) {}
 
 #ifdef CVC5_NEED_INT64_T_OVERLOADS
-  Integer(int64_t z) : d_value(static_cast<long>(z)) {}
-  Integer(uint64_t z) : d_value(static_cast<unsigned long>(z)) {}
+  Integer(int64_t z) : d_value(z) {}
```

Comment:
So CLN didn't need the casts to begin with?

Reply:
The short answer: yes

My guess is that whoever wrote the CLN implementation in CVC5, started by copying the GMP implementation -- when it came to the sections inside of `CVC5_NEED_INT64_T_OVERLOADS`, they didn't check if the casts were required or not (they're not). They likely didn't check because hitting that code is hard (only Windows or 32-bit Linux).

In summary: anywhere a term was casted to `long` in GMP was incorrect (now fixed) and anywhere a term was casted to `long` in CLN was unnecessary (now also fixed).

I had attempted to capture some of these points in the (updated) PR description; take a look and let me know if you feel I cover the intent of these changes.

Reply:
> My guess is that whoever wrote the CLN implementation in CVC5, started by copying the GMP implementation -- when it came to the sections inside of `CVC5_NEED_INT64_T_OVERLOADS`, they didn't check if the casts were required or not (they're not). They likely didn't check because hitting that code is hard (only Windows or 32-bit Linux).

I was wrong:

* https://github.com/cvc5/cvc5/commit/94b685e149119bbe75266481f6de3a0915e7c10a

I have no idea why those casts are there for CLN (maybe CLN didn't used to support `long long`?); maybe Morgan just made the code be consistent between GMP and CLN, without realising it wasn't required for CLN?